.github/workflows/canary-release.yml

```diff
@@ -0,0 +1,97 @@
+        uses: martinbeentjes/npm-get-version-action@master
+      - name: remove old draft releases
+        run: gh release list | grep -e "Draft.*v${{ steps.package-version.outputs.current-version }}" | awk '{print $3}' | xargs -r gh release delete
+      - name: create new draft release
+        uses: marvinpinto/action-automatic-releases@latest
+        with:
+          repo_token: ${{ secrets.GH_RELEASE_DEPLOYMENT_TOKEN }}
+          automatic_release_tag: v${{ steps.package-version.outputs.current-version }}
+          prerelease: true
+          draft: false
+          title: ${{ steps.package-version.outputs.current-version }}
+
+  build_release:
+    needs: [create_canary_release]
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+        include:
+          - os: macos-11
+            cert_key: APPLE_DEVELOPER_SIGNING_CERTIFICATE
+            cert_password_key: APPLE_CERTIFICATE_PASSWORD
+            apple_id_key: APPLE_ID
+            apple_id_password_key: APPLE_ID_PASSWORD
+          - os: windows-latest
+            cert_key: WINDOWS_CSC_CERTIFICATE
+            cert_password_key: WINDOWS_CSC_PASSWORD
+
+    runs-on: ${{ matrix.os }}
+
+    env:
+      CSC_LINK: ${{ secrets[matrix.cert_key] }}
+      CSC_KEY_PASSWORD: ${{ secrets[matrix.cert_password_key] }}
+      APPLE_ID: ${{ secrets[matrix.apple_id_key] }}
+      APPLE_ID_PASSWORD: ${{ secrets[matrix.apple_id_password_key] }}
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 16.15.0
+          cache: "npm"
+      - name: read version from package.json
+        id: package-version
+        uses: martinbeentjes/npm-get-version-action@master
+      - name: install Linux dev tools
+        if: matrix.os == 'ubuntu-latest'
+        run: |
+          sudo apt update -y
+          sudo apt install -y libudev-dev
+      - name: create NPM config file
+        run: echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_AUTH_TOKEN }}" > ./.npmrc
+      - run: npm run setup:ci
+      - run: npm run compile
+      - run: npm run test:unit:main -- --testTimeout=500 --verbose
+      - run: npm run bundle
+      - name: Sleep for 4 seconds
+        shell: bash
+        run: sleep 4
+      - run: npm run publish
+
+  bump_release:
+    needs: [build_release]
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Bump version
+        id: package_version
+        uses: KageKirin/bump-node-package-version@v1.0.0
+        with:
+          regex: /(\d)\.(\d)\.[\d]-canary\.(\d)/
+          patch: true
+
+      - name: Commit new version
+        run: |
+          git commit -am "CI: bump version to ${{ steps.package_version.outputs.version }}"
+          git tag -m "CI: create new tag" v${{ steps.package_version.outputs.version }}
+          git push --follow-tags https://${{ env.GH_TOKEN }}@github.com/frame-labs/frame-canary
```

Comment:
I think this is the part that we need, but it should probably be a separate action that's triggered by a publish of a release